web/crux/src/app/registry/registry-clients/registry-api-client.ts

```diff
@@ -1,6 +1,30 @@
+import {
+  CruxBadRequestException,
+  CruxInternalServerErrorException,
+  CruxUnauthorizedException,
+} from 'src/exception/crux-exception'
 import { RegistryImageTags } from '../registry.message'
 
 export interface RegistryApiClient {
-  catalog(text: string, take: number): Promise<string[]>
+  catalog(text: string): Promise<string[]>
   tags(image: string): Promise<RegistryImageTags>
 }
+
+export const getRegistryApiException = (res: Response, endpoint: string) => {
```

Comment:
Is this the right place for this? We have some error handling and exception definition files, and I believe it would be more suitable to have them there.

Reply:
You are right, I could move it to the exceptions folder.